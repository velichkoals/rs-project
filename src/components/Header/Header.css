.header__item {
    background-color: #f7f7f7;
    color: #6752dc;
    padding: 0 20px;
    text-decoration: none;
    font-weight: 500;
}
.header__item:hover {
    text-decoration: underline;
    color: #9a4fe1;
    transition: .3s ease-in-out color;
}
.header-wrapper{
    position: relative;
}

.header__item__profile {
    position: absolute;
    right: 4rem;
}