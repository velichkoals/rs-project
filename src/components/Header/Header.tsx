import React from "react";
import { Link } from "react-router-dom";
import "./Header.css";

const Header = () => {
  return (
    <div>
      <div className="header-wrapper">
        <Link data-testid="home-link" to="/" className="header__item">
          Home
        </Link>
        <Link data-testid="about-link" to="/about" className="header__item">
          About Us
        </Link>
        <Link
          data-testid="profile-link"
          to="/profile"
          className="header__item header__item__profile"
        >
          Profile
        </Link>
      </div>
    </div>
  );
};
export default Header;
