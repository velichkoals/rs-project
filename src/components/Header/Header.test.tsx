import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Header from "./Header";

it("Header renders", () => {
  const header = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );
  expect(header).toMatchSnapshot();
});
