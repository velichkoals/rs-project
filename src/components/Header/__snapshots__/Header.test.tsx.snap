// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="header-wrapper"
        >
          <a
            class="header__item"
            data-testid="home-link"
            href="/"
          >
            Home
          </a>
          <a
            class="header__item"
            data-testid="about-link"
            href="/about"
          >
            About Us
          </a>
          <a
            class="header__item header__item__profile"
            data-testid="profile-link"
            href="/profile"
          >
            Profile
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="header-wrapper"
      >
        <a
          class="header__item"
          data-testid="home-link"
          href="/"
        >
          Home
        </a>
        <a
          class="header__item"
          data-testid="about-link"
          href="/about"
        >
          About Us
        </a>
        <a
          class="header__item header__item__profile"
          data-testid="profile-link"
          href="/profile"
        >
          Profile
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
